### PR TITLE
fix(eslint-plugin): [unbound-method] ignore assignments _to_ methods

### LINKED ISSUE #1256: [unbound-method] should ignore assignments
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/unbound-method": "error"
  }
}
```

```TS
// PLEASE IGNORE, BAD REPRO
// let obj = {};
// obj.fn = (function() {}).bind(obj);

class A {
  foo(): void {
    this.foo = undefined;
  }
}
```

**Expected Result**

No error.

**Actual Result**

'unbound-method' complains about ~`obj.fn`~ `this.foo`.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.7.0` |
| `@typescript-eslint/parser`        | `2.7.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12.12.0` |
| `npm`                              | `6.11.3` |

### PR DESCRIPTION
**Pull request by Zzzen**
closes #1256

### CODE REVIEW

File: packages/eslint-plugin/src/rules/unbound-method.ts [hunk 1]
```diff
@@ -241,6 +241,9 @@ function isSafeUse(node: TSESTree.Node): boolean {
     case AST_NODE_TYPES.BinaryExpression:
       return ['instanceof', '==', '!=', '===', '!=='].includes(parent.operator);
 
+    case AST_NODE_TYPES.AssignmentExpression:
+      return parent.operator === '=';
```
Comment (bradzacher) [inline packages/eslint-plugin/src/rules/unbound-method.ts]: Could this also check that the node is on the correct side?
i.e.

```ts
class Foo { unbound() {} }
const instance = new Foo();

let x;

x = instance.unbound; // THIS SHOULD ERROR
instance.unbound = x; // THIS SHOULD NOT
```

Comment (Zzzen) [reply packages/eslint-plugin/src/rules/unbound-method.ts]: Holy crap! How can you be so considerate!

Comment (Zzzen) [reply packages/eslint-plugin/src/rules/unbound-method.ts]: Tests have been updated. It would be great if you could review again.

Comment (Zzzen) [reply packages/eslint-plugin/src/rules/unbound-method.ts]: PTAL

**Comment by Zzzen** [comment]
006dMd5bgy1fkvshz132ag305k05knpf

anyway, fixed it.

**Pull request merged**